include/pybind11/numpy.h

```diff
@@ -1083,7 +1083,7 @@ inline PYBIND11_NOINLINE void register_structured_dtype(
     // - https://github.com/numpy/numpy/pull/7798
     // Because of this, we won't use numpy's logic to generate buffer format
     // strings and will just do it ourselves.
-    std::vector<field_descriptor> ordered_fields(fields);
+    std::vector<field_descriptor> ordered_fields(*fields);
```

Comment:
You should be able to use just `std::move(fields)` here instead of `*fields` (to avoid a vector copy).

Reply:
Hmm I guess it was actually always copying them but yea -- `fields` is not used past this point so we could just move them; fixed.